lib/Core/Executor.cpp

```diff
@@ -1321,26 +1313,44 @@ Executor::toConstant(ExecutionState &state,
   if (ConstantExpr *CE = dyn_cast<ConstantExpr>(e))
     return CE;
 
-  ref<ConstantExpr> value;
-  bool success =
-      solver->getValue(state.constraints, e, value, state.queryMetaData);
-  assert(success && "FIXME: Unhandled solver failure");
-  (void) success;
+  ref<ConstantExpr> ce;
+  ref<Expr> value = nullptr;
+  if (auto found = seedMap.find(&state); found != seedMap.end())
+    value = getValueFromSeeds(found->second, e);
+  /* If no seed evaluation results in a constant, call the solver */
+  if (!value || !(ce = dyn_cast<ConstantExpr>(value))) {
```

Comment:
```suggestion
  ref<Expr> value;
  if (auto found = seedMap.find(&state); found != seedMap.end())
    value = getValueFromSeeds(found->second, e);
  /* If no seed evaluation results in a constant, call the solver */
  ref<ConstantExpr> cvalue = llvm::dyn_cast_or_null<ConstantExpr>(value);
  if (!cvalue) {
```

Reasoning: 
- the variable name `ce` clashes with the previously used `CE` (which is the constant variant of `e`). Following the same naming scheme gives `cvalue` as the constant variant of `value`.
- The variable should be declared and initialized at once, where possible.
  - `dyn_cast_or_null` significantly simplifies the logic while preserving semantics
- `ref<T>` default initializes to an empty reference, so explicitly setting it to `nullptr` is not necessary